wpical/BUILD.bazel

```diff
@@ -0,0 +1,248 @@
+load("//shared/bazel/rules/gen:gen-resources.bzl", "generate_resources")
+load("//shared/bazel/rules/gen:gen-version-file.bzl", "generate_version_file")
+
+pkg_files(
+    name = "licenses",
+    srcs = [
+        "WPICalThirdPartyNotices.txt",
+    ],
+)
+
+generate_resources(
+    name = "generate-resources",
+    namespace = "wpical",
+    prefix = "WPI",
+    resource_files = glob(["src/main/native/resources/*"]),
+)
+
+generate_version_file(
+    name = "generate-version",
+    output_file = "WPILibVersion.cpp",
+    template = "src/main/generate/WPILibVersion.cpp.in",
+)
+
+cc_library(
+    name = "headers",
+    hdrs = glob([
+        "src/main/native/include/**/*",
+    ]),
+    strip_include_prefix = "src/main/native/include",
+    deps = [
+        ":headers-libdogleg",
+        ":headers-mrcal",
+        ":headers-mrcal-generated",
+        ":headers-mrcal-java",
+    ],
+)
+
+cc_library(
+    name = "headers-libdogleg",
+    hdrs = glob([
+        "src/main/native/thirdparty/libdogleg/include/**/*",
+    ]),
+    strip_include_prefix = "src/main/native/thirdparty/libdogleg/include",
+)
+
+cc_library(
+    name = "headers-mrcal-generated",
+    hdrs = glob([
+        "src/main/native/thirdparty/mrcal/generated/**/*",
+    ]),
+    strip_include_prefix = "src/main/native/thirdparty/mrcal/generated/",
+)
+
+cc_library(
+    name = "headers-mrcal",
+    hdrs = glob([
+        "src/main/native/thirdparty/mrcal/include/**/*",
+    ]),
+    strip_include_prefix = "src/main/native/thirdparty/mrcal/include/",
+)
+
+cc_library(
+    name = "headers-mrcal-java",
+    hdrs = glob([
+        "src/main/native/thirdparty/mrcal_java/include/**/*",
+    ]),
+    strip_include_prefix = "src/main/native/thirdparty/mrcal_java/include/",
+)
+
+unix_copts = [
+    "-Wno-pedantic",
+    "-Wno-format-nonliteral",
+    "-Wno-unused-variable",
+    "-Wno-unused-function",
+    "-Wno-sign-compare",
+]
+
+osx_copts = unix_copts
+
+copts = select({
+    "@platforms//os:linux": unix_copts + [
+        "-Wno-maybe-uninitialized",
+    ],
+    "@platforms//os:osx": osx_copts,
+    "@platforms//os:windows": [
+        "/wd4047",
+        "/wd4098",
+        "/wd4267",
+    ],
+})
+
+unix_cxxopts = [
+    "-Wno-missing-field-initializers",
+    "-Wno-pedantic",
+    "-fpermissive",
+    "-Wno-deprecated-declarations",
+    "-Wno-return-type",
+    "-Wno-missing-braces",
+    "-Wno-null-conversion",
+    "-Wno-unused-but-set-variable",
+]
+
+osx_cxxopts = unix_cxxopts + [
+    "-Wno-unused-variable",
+    "-Wno-unused-function",
+    "-Wno-sign-compare",
+    "-Wno-sometimes-uninitialized",
+]
+
+cxxopts = select({
+    "@platforms//os:linux": unix_cxxopts + [
+        "-Wno-deprecated-enum-enum-conversion",
+    ],
+    "@platforms//os:osx": osx_cxxopts,
+    "@platforms//os:windows": [
+        "/wd4068",
+        "/wd4101",
+        "/wd4200",
+        "/wd4576",
+        "/wd4715",
+    ],
+})
+
+mac_linkopts = [
+    "-framework",
```

Comment:
Are these framework dependencies here because `//wpigui` and/or opencv depend on them? Can we fix that build to correctly declare the frameworks it links to in the `objc_library`?

Reply:
I think that should only be if you are using `dynamic_deps`, which this isn't. I would think for a standard library and a standard binary it should pull them in.

I couldn't find the bug linked in the `main`, but one of my PR's has it. https://github.com/bazelbuild/bazel/issues/21893